packages/image/README.md

```diff
@@ -0,0 +1,108 @@
+# `@sveltejs/image`
+
+**WARNING**: This package is experimental. It uses pre-1.0 versioning and may introduce breaking changes with every minor version release.
+
+This package aims to bring a plug&play image component to SvelteKit that is opinionated enough so you don't have to worry about the details, yet flexible enough for more advanced use cases or tweaks. It uses the `srcset` and `sizes` attributes of the `img` tag to provide resized images suitable for various device sizes, which for example results in smaller images downloaded for mobile.
+
+## Setup
+
+Install:
+
+```bash
+npm install --save @sveltejs/image
+```
+
+Adjust `vite.config.js`:
+
+```diff
++import { vitePluginSvelteImage } from '@sveltejs/image/vite';
+import { sveltekit } from '@sveltejs/kit/vite';
+import { defineConfig } from 'vite';
+
+export default defineConfig({
+	plugins: [
++		vitePluginSvelteImage({ providers: { default: '@sveltejs/image/providers/<choose one>' } }),
+		sveltekit()
+	]
+});
+```
+
+> `<choose one>` refers to chosing one of the ready-to-use providers. We plan to add more providers over time. You can create your own by creating a JavaScript with a `export function getURL({ src, width, height }): string` function inside.
+
+In case of Vercel, adjust `svelte.config.js`:
+
+```diff
+import adapter from '@sveltejs/adapter-vercel';
+
+/** @type {import('@sveltejs/kit').Config} */
+const config = {
+	kit: {
+-		adapter: adapter()
++		adapter: adapter({ images: true })
+	}
+};
+
+export default config;
+```
+
+## Usage
+
+### When using one of the providers, i.e. an image CDN:
+
+```svelte
+<script>
+	import Image from '@sveltejs/image';
+</script>
+
+<Image src="/path/to/your/image.jpg" width={1200} height={1800} alt="An alt text" />
```

Comment:
This cannot be cached. If you `import` it then Vite will compute a hash giving it a unique URL that you can set far forwards expires headers one. I think we should still encourage images to be imported for dynamic CDNs.

You could set a project-level default for static vs dynamic and then opt-in/out with a query parameter that triggers the `include`/`exclude` of `vite-imagetools`

Reply:
".. that you can set far forwards expires headers one" - I don't understand what you mean by that

Reply:
"one" was a typo and was supposed to be "on". What I mean is setting `'cache-control': 'public, immutable, max-age=31536000'`

Reply:
When we talked this morning you said `/path/to/your/image.jpg` is something that would probably not exist locally. Do we need some documentation around how to deploy it? Or maybe some best practices like creating an `/immutable` path that can be cached and images outside of that would not be cached or something? Maybe with examples for the built-in providers?

Reply:
Not sure I follow. What I meant is that if your `src` path starts with a `/` or with `http(s)://` it indicates that you're not importing some local image file on disk but instead want to reference something through the URL. What that means with regards to caching depends on where the image is stored. I wouldn't worry about that for now, we can document that later (since it's only a documentation thing, not an implementation issue).

Reply:
using `/favicon.ico` resolves to `/static/favicon.ico` locally. so it could be confusing if some paths lead to local sources and others to a cdn url. Would an alias work? eg. $cdn where $cdn is a configured value from svelte/image/vite ?

Reply:
problem solved

```suggestion
<Image src="https://example.com/path/to/your/image.jpg" width={1200} height={1800} alt="An alt text" />
```